simple_mediation.R

```diff
@@ -1,86 +1,58 @@
-neProsocial <- neModel(prosocial ~ SEP0 + SEP1 + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race, expData = expData, se = "robust")
-summary(neProsocial)
-str(neProsocial)
-
-#I trired to "unlist" the results but this did not solve the error
-#unlist(neProsocial) -> neProsocial2
-head(neProsocial)
-
-effdecomp <- neEffdecomp(neProsocial)
-summary(effdecomp)
-
-summary(expData)
-head(expData)
-expData2 <- neImpute(prosocial ~ SEP + model3_trajectory_assignments + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_ethnicity + father_ethnicity, family = poisson(link = "log"), data = df_males_na_removed)
\ No newline at end of file
+library(rmarkdown)
+
+source("./load_data_by_sex.R")
+source("./munging.R")
+male_data()
+munge("male")
+female_data()
+munge("female")
+
+#Run loop for all socio-emotional variables
+for (x in 16:ncol(df_male)) {
+  name <- colnames(df_male[,x])
+  assign("objectname", (paste(name,"~","SEP + model3_trajectory_assignments + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race")))
+  expData <- assign(paste0("expData_", name), neImpute(objectname, data = df_male))
+  assign("model", (paste(name,"~",("SEP0 + SEP1 + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race"))))
+  outcome_results <- assign(paste0("ne_", name), neModel(model, expData = expData, se = "robust"))
+  assign(paste0("male_effdecomp_", name), envir = globalenv(), neEffdecomp(outcome_results))
+}
+#Summary of results in "results__male_mediation.Rmd" 
+
+for (x in 16:ncol(df_female)) {
+  name <- colnames(df_female[,x])
+  assign("objectname", (paste(name,"~","SEP + model4_trajectory_assignments + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race")))
+  expData <- assign(paste0("expData_", name), neImpute(objectname, data = df_female))
+  assign("model", (paste(name,"~",("SEP0 + SEP1 + Remoteness + Indigenous + adv + dis + resources + education_occupation + mother_home + father_home + mental_health + mother_race + father_race"))))
+  outcome_results <- assign(paste0("ne_", name), neModel(model, expData = expData, se = "robust"))
+  assign(paste0("female_effdecomp_", name), envir = globalenv(), neEffdecomp(outcome_results))
+}
+
+#See RMD file "simple_mediation_results.rmd" for summaries of all effdecomp objects 
+
+
+
+#I tried to make a function but I couldn't get it to work! I couldn't figure it out but it kept returning erros out on the last line. No clue why.  
```

Comment:
Hard to say. Nothing is obviously wrong and I'd need the error message to be more help.

I would remove all of this though.